### ISSUE TITLE
[BUG] Cython 3.x regression with dunder named properties

### ISSUE THREAD

**Issue by pythonengineer** [cython/cython#5674]
### Describe the bug

If a class property is named with double underscores, an AttributeError will be observed when attempting to access that property.

### Code to reproduce the behaviour:

```cython
cdef class Object:

    property __example:

        def __get__(self):
            return 1

    def __init__(self):
        print(self.__example)
```

### Expected behaviour

On Cython 2.x, it behaves as expected with no AttributeError.

### OS

Windows

### Python version

3.9

### Cython version

3.0.2

**Comment by da-woods** [comment]
Yeah it looks like the name `__example` doesn't get mangled when declaring the property but does when trying to look it up.

I suspect it'll work if you use `@property` (the old property syntax predates that). But even so it should work here.

**Comment by pythonengineer** [comment]
With my testing, the same issue occurs with the `@property` syntax.
